experiments/AtmosGCM/solid-body-rotation.jl

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env julia --project
+using ClimateMachine
+ClimateMachine.init(parse_clargs = true)
+
+using ClimateMachine.Atmos
+using ClimateMachine.Orientations
+using ClimateMachine.ConfigTypes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.SystemSolvers: ManyColumnLU
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.VariableTemplates
+using ClimateMachine.Thermodynamics: air_density, total_energy
+
+using LinearAlgebra
+using StaticArrays
+using Test
+
+using CLIMAParameters
+using CLIMAParameters.Planet: day, planet_radius
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+function init_solid_body_rotation!(problem, bl, state, aux, coords, t)
+    FT = eltype(state)
+
+    # The initial state is in hydrostatic balance
+    # It is chosen to be the same as reference state
+    # But we should expect the flow to stay at rest
+    # with any hydrostatic initial state
+    temp_profile_init =
+        DecayingTemperatureProfile{FT}(param_set, FT(290), FT(220), FT(8e3))
+    z = altitude(bl.orientation, bl.param_set, aux)
+    T₀, p = temp_profile_init(bl.param_set, z)
+    ρ = air_density(bl.param_set, T₀, p)
+    e_pot = gravitational_potential(bl.orientation, aux)
+    e_kin = FT(0)
+
```

Comment:
This is not exactly the same as the reference state because in the reference state, the density is calculated discretely from hydrostatic balance and pressure, and the temperature is determined to be consistent with that density. See https://github.com/CliMA/ClimateMachine.jl/blob/8a97053e7acaa58d9e67e85abf2d07e1883d6255/src/Atmos/Model/ref_state.jl#L85-L107.

We should make the result of the reference state calculations available to initialize the model. Perhaps @charleskawczynski can help making that possible. An option might be to move the calculations of density, temperature etc. from `ref_state` to `TemperatureProfiles`, and then use those results both for initialization and the reference state.

Reply:
Thanks, Tapio. @charleskawczynski What I would like to have here is to have `state.ρ = aux.ref_state.ρ` and `state.ρe = aux.ref_state.ρe` but use `TemperatureProfiles` instead of `ref_state`. What would be the best way to achieve this?

Reply:
This is actually why we reverted this back. We quickly discovered that it was a bit more involved since the discrete part is living inside this function and requires a discrete pressure gradient that I am note sure is available to the init function.

Reply:
@charleskawczynski, my intuition as a user is that when I create a hyrostatic state, I will have easy access to a set of state variables (dry or moist). Then it would be easy to use that kind of object in the init function.

Reply:
We have access to the ref state vars in `init_state_prognostic!`: `aux.ref_state.x`. IIUC, I think that'd be fine

Reply:
What's the benefit of computing / recomputing this from `TemperatureProfiles`?

Reply:
@charleskawczynski What we need is a little different: We want to be able to set up a hydrostatic reference and *also* a hydrostatic initial state, which may differ from the reference state but is computed in the same way (i.e., is in discrete hydrostatic balance).

What would make sense to me is to have a function that computes a hydrostatic state given input parameters in the same way the reference state is currently computed. This could be an augmented version of the current `TemperatureProfiles` (`HydrostaticProfiles`?). We could then use that function both to set up a reference state (with one set of parameters) and an initial state (with another set of parameters).

Reply:
I see.

> which may differ from the reference state

For example, using different input parameters to `DecayingTemperatureProfile`?

Reply:
Yes, exactly.